### PR TITLE
chore: custom pepr image for ci testing

### LINKED ISSUE #760: allow custom dev image for pepr for release candidate testing against UDS Core
### Is your feature request related to a problem? Please describe.

We need to be able to run uds packages with a pepr release candidate image in CI.

**Background:** 
When Pepr updates functionality, although they are heavily tested, we don't always know how they will affect UDS Core. Pepr wants to run a compatibility test in CI. 

### Describe the solution you'd like

- **Given** a PR in issued to Pepr that may affect UDS Core
- **When** the Pepr image is built, a uds run command of some sort is run in Pepr CI
- **Then** the packages are fetched to make sure they all reconciled successfully. 

### Describe alternatives you've considered
(optional) A clear and concise description of any alternative solutions or features you've considered.

### PR DESCRIPTION
**Pull request by cmwylie19**
## Description

This allows a custom release candidate image of pepr to be run with UDS Core for CI testing 

## Related Issue

Fixes #760 

Relates to #

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Other (security config, docs update, etc)

## Checklist before merging

- [x] Test, docs, adr added or updated as needed
- [x] [Contributor Guide](https://github.com/defenseunicorns/uds-template-capability/blob/main/CONTRIBUTING.md) followed

### CODE REVIEW

File: tasks/create.yaml [hunk 1]
```diff
@@ -60,7 +60,17 @@ tasks:
     actions:
       - description: "Build the UDS Core Pepr Module"
         cmd: |
-          CUSTOM_PEPR_IMAGE=$( [ "${FLAVOR}" = "registry1" ] && echo "--custom-image ${REGISTRY1_PEPR_IMAGE}" ) || CUSTOM_PEPR_IMAGE=""
+          if [ "${FLAVOR}" = "registry1" ] ; then
```
Comment (rjferguson21) [inline tasks/create.yaml]: IMO we should let `PEPR_CUSTOM_IMAGE` override the pepr image even if the flavor is registry1.

Comment (btlghrants) [reply tasks/create.yaml]: We don't feel too strongly about what should override what, so deferring to you!

PR updated to allow PEPR_CUSTOM_IMAGE to override FLAVOR below. :+1:

File: tasks/create.yaml [hunk 2]
```diff
@@ -60,7 +60,17 @@ tasks:
...
+            PEPR_OVERRIDE_IMAGE="--custom-image ${REGISTRY1_PEPR_IMAGE}"
+
+          if [ -n "${PEPR_CUSTOM_IMAGE}" ] ; then
+            # e.g. PEPR_CUSTOM_IMAGE="pepr:dev uds run slim-dev"
+            PEPR_OVERRIDE_IMAGE="--custom-image ${PEPR_CUSTOM_IMAGE}"
+
+          else
+            PEPR_OVERRIDE_IMAGE=""
+          fi
```
Comment (mjnagel) [inline tasks/create.yaml]:
```suggestion
          if [ -n "${PEPR_CUSTOM_IMAGE}" ] ; then
            # e.g. PEPR_CUSTOM_IMAGE="pepr:dev uds run slim-dev"
            PEPR_OVERRIDE_IMAGE="--custom-image ${PEPR_CUSTOM_IMAGE}"
          elif [ "${FLAVOR}" = "registry1" ] ; then
            PEPR_OVERRIDE_IMAGE="--custom-image ${REGISTRY1_PEPR_IMAGE}"
          else
            PEPR_OVERRIDE_IMAGE=""
          fi
```

I think this fixes the syntax issues and sets the proper precedence.

Comment (mjnagel) [reply tasks/create.yaml]: Might also request that we add `PEPR_OVERRIDE_IMAGE` as a variable to the top of this task, with a description of its usage if possible.

Comment (btlghrants) [reply tasks/create.yaml]: Pfft, yeah... was rushed when pushing the update to this -- meant to make those 3 independent if/fi blocks that all just overrode each other top-down -- dunno what I was thinking with that incomplete condition.  Oof.

I think what you have works just fine too though.

**Review by mjnagel — APPROVED** [review]

**Pull request closed without merging**